__tests__/client/service-worker/events/caching.spec.js

```diff
@@ -388,8 +388,10 @@ describe('caching', () => {
     }, Promise.resolve([]));
 
     expect(fetch).toHaveBeenCalledTimes(3);
-    events.forEach((event) => {
-      expect(event.waitUntil).toHaveBeenCalledTimes(3);
+    events.forEach((event, index) => {
+      // on the third call, we expect to waitUntil the original cache item was removed
+      // as an additional step when invalidating
+      expect(event.waitUntil).toHaveBeenCalledTimes(index === 2 ? 4 : 3);
```

Comment:
This isnt clear to me. Can you explain this in more detail why the third call is important or what script the 3rd call is referring to?

Reply:
The reason for the difference in the third call is that `event.waitUntil` is called an extra time to wait for the invalidation to delete the item. I'll add comments, and maybe rethink the test to make it clearer.